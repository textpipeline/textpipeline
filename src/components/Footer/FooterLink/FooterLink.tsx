import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import SvgIcon from '@material-ui/core/SvgIcon';
import TwitterIcon from '@material-ui/icons/Twitter';
import React from 'react';

import ExternalLink from '../../ExternalLink';

export interface FooterLinkProps {
  readonly icon: string;
  readonly text: string;
  readonly href: string;
}

const linkColor = 'textPrimary';

const iconMapping: Record<string, typeof SvgIcon> = {
  twitter: TwitterIcon,
};

const FooterLink: React.FC<FooterLinkProps> = ({ icon, text, href }) => {
  const Icon = iconMapping[icon];
  return (
    <ListItem component={ExternalLink} href={href} color={linkColor}>
      {Icon && (
        <ListItemIcon>
          <Icon />
        </ListItemIcon>
      )}
      <ListItemText>{text}</ListItemText>
    </ListItem>
  );
};

export default FooterLink;
