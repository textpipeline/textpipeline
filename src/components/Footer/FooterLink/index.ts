export { default } from './FooterLink';
export * from './FooterLink';
