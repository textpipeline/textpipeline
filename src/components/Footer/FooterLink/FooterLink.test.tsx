import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import FooterLink from './FooterLink';

describe('FooterLink', () => {
  let rendered: ShallowWrapper;

  describe('with a valid icon', () => {
    beforeEach(() => {
      rendered = shallow(<FooterLink icon="twitter" text="Testing" href="https://example.com/foo" />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('with an invalid icon', () => {
    beforeEach(() => {
      rendered = shallow(<FooterLink icon="boooooooop" text="Testing" href="https://example.com/foo" />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });
});
