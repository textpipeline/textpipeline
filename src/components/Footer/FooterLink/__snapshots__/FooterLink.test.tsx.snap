// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLink with a valid icon renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  color="textPrimary"
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  href="https://example.com/foo"
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <TwitterIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))>
    Testing
  </WithStyles(ForwardRef(ListItemText))>
</WithStyles(ForwardRef(ListItem))>
`;

exports[`FooterLink with an invalid icon renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  color="textPrimary"
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  href="https://example.com/foo"
>
  <WithStyles(ForwardRef(ListItemText))>
    Testing
  </WithStyles(ForwardRef(ListItemText))>
</WithStyles(ForwardRef(ListItem))>
`;
